### ISSUE TITLE
Remove excess calls to insertionSort

### ISSUE THREAD

**Issue by cmaceves** [andersen-lab/ivar#147]
In the code for primer trimming insertionSort gets called every time get_overlapping_primers is called, creating a slowdown

```
std::vector<primer> test = insertionSort(primers, primers.size());
```

This should only be called once.

**Comment by tseemann** [comment]
@cmaceves we switched to ivar 1.3.2 last week and noticed it was taking 10x (!) longer to run on our 384 samples (~1 M reads each).

Would this bug explain the slowdown in ivar trim?

**Comment by cmaceves** [comment]
hi @tseemann! that's correct, it is causing the slowdown. we're aiming for a fast turn around time on 1.3.3 to address that oversight.

**Comment by cmaceves** [comment]
addressed in PR #155
